code/game/machinery/_machinery.dm

```diff
@@ -198,14 +198,16 @@
 	end_processing()
 	dump_inventory_contents()
 
-	if (!isnull(component_parts))
+	if(!isnull(component_parts))
 		// Don't delete the stock part singletons
 		for (var/atom/atom_part in component_parts)
 			qdel(atom_part)
 		component_parts.Cut()
 		component_parts = null
 
-	QDEL_NULL(circuit)
+	if(!isnull(circuit))
```

Comment:
you do not need this check, QDEL_NULL(null) is not only benign but something lemon wants to make faster in the future